pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -189,6 +196,235 @@ func getV1Registries(config *tomlConfig) ([]Registry, error) {
 	return registries, nil
 }
 
+// registryIsMatch returns true if flags in the registry entry indicate that
+// its URL might not be used as-is, but will have URLs matched against it in
+// order to selectively apply configuration flags to an unlisted URL.
+func registryIsMatch(reg Registry) bool {
+	return reg.Insecure || reg.Blocked
+}
+
+// registryMatches returns true if the candidate location matches the pattern.
+// Recognized patterns include:
+// * IPv4[/class][:port][/path]
+// * IPv6[/class][:port][/path]
```

Comment:
(For better or worse, the `[ipv6]` syntax is not valid in Docker references; it will never match any acceptable input.)